### DISCUSSION TITLE
Hi, a question about UI frontstageprovider

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#849] [Q&A] [answered]
Hi, I place custom controls in centerright or centerleft of frontstageprovider, but I can't drag the window of the control freely to make it bigger. Sometimes it may affect the visual effect. Is there any way to achieve it? thank you.

**Comment by bsteinbk** [comment, marked as answer]
There toolbars are limited to the area they can occupy.  If all tool buttons cannot fit into the available zone, an overflow button should typically be created to provide access to all defined tools. @shenshengyi can you provide a screen shot that shows the problem you are seeing?

**Reply by shenshengyi** [reply]
> There toolbars are limited to the area they can occupy. If all tool buttons cannot fit into the available zone, an overflow button should typically be created to provide access to all defined tools. @shenshengyi can you provide a screen shot that shows the problem you are seeing?

![image](https://user-images.githubusercontent.com/26344907/109243334-b5867180-7817-11eb-9350-cd7015955222.png)

I want to enlarge the window area on the left, even to the far right (because there may be tables with many columns).

However, it can only be raised to a certain limit now.

What can I do? Thank you

**Reply by bsteinbk** [reply]
@shenshengyi  what you are showing is a widget within a zone in the "Ninezone" UI.  The UX design for zones does not allow it to extend horizontally outside its zone limits. There are at least two options for displaying controls across a wider area.

The first is to define a content area to hold the controls. Below is an example of a table defined with this method, and here is a [link](https://github.com/imodeljs/imodeljs/blob/4b499c67e335793252b9cf50d60d77f1cc0a0f2d/test-apps/ui-test-app/src/frontend/appui/frontstages/Frontstage3.tsx#L47) to the code example.

![image](https://user-images.githubusercontent.com/65047615/109342226-8bde4200-7839-11eb-9277-64875beec782.png)

The second option is to define a bottom panel for the UI controls. The following example shows a bottom panel with a player component. This is a good option for displaying very wide content. Here is a [link](https://github.com/imodeljs/imodeljs/blob/4b499c67e335793252b9cf50d60d77f1cc0a0f2d/test-apps/ui-test-app/src/frontend/appui/frontstages/Frontstage1.tsx#L220) to the code that specifies the panel.

![image](https://user-images.githubusercontent.com/65047615/109343029-b5e43400-783a-11eb-8af7-f2c3822464f9.png)

**Reply by shenshengyi** [reply]
Thank you very much. Your answers have solved my problems and other doubts.
